diff -urN bin.orig/activate bin/activate
--- bin.orig/activate  2018-12-27 14:55:13.916461020 +0900
+++ bin/activate       2018-12-27 20:38:35.223248728 +0900
@@ -30,6 +30,15 @@
         unset _OLD_VIRTUAL_PS1
     fi

+    # Unset exported dev-env variables
+    pushd ${DEVENV_PATH} > /dev/null
+    unset `make env | awk -F= '{print $1}'`
+    popd > /dev/null
+
+    # Unset external env variables
+    declare -f env_deactivate > /dev/null && env_deactivate
+    declare -f venv_deactivate > /dev/null && venv_deactivate
+
     unset VIRTUAL_ENV
     if [ ! "${1-}" = "nondestructive" ] ; then
     # Self destruct!
@@ -47,6 +56,11 @@
 PATH="$VIRTUAL_ENV/bin:$PATH"
 export PATH

+# Set external variables
+if [ -f ${VIRTUAL_ENV}/bin/environment.sh ] ; then
+    . ${VIRTUAL_ENV}/bin/environment.sh
+fi
+
 # unset PYTHONHOME if set
 if ! [ -z "${PYTHONHOME+_}" ] ; then
     _OLD_VIRTUAL_PYTHONHOME="$PYTHONHOME"
