--- activate.orig	2023-11-01 16:36:41.273318317 +0000
+++ activate	2023-11-01 16:29:45.610264478 +0000
@@ -27,6 +27,10 @@
         unset _OLD_VIRTUAL_PS1
     fi

+    # Unset external env variables
+    declare -f env_deactivate > /dev/null && env_deactivate
+    declare -f venv_deactivate > /dev/null && venv_deactivate
+
     unset VIRTUAL_ENV
     unset VIRTUAL_ENV_PROMPT
     if [ ! "${1:-}" = "nondestructive" ] ; then
@@ -45,6 +56,11 @@
 PATH="$VIRTUAL_ENV/bin:$PATH"
 export PATH
 
+# Set external variables
+if [ -f ${VIRTUAL_ENV}/bin/environment.sh ] ; then
+    . ${VIRTUAL_ENV}/bin/environment.sh
+fi
+
 # unset PYTHONHOME if set
 # this will fail if PYTHONHOME is set to the empty string (which is bad anyway)
 # could use `if (set -u; : $PYTHONHOME) ;` in bash
